### ISSUE TITLE
Download speed is extremely slow when the default UA is applied and no referer

### ISSUE THREAD

**Issue by fireattack** [instaloader/instaloader#1233]
**Describe the bug**

When downloading the media, the download speed is very slow, as low as 50KB/s.

**To Reproduce**
```sh
instaloader -- -CRjFJnKJVRe
```

**Expected behavior**
The download speed should be relatively fast.

**Instaloader version**

`master`

**Additional context**
After some debugging, I found this problem only happens when the default UA `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36` is used and no referer of `https://www.instagram.com/` is applied.

Removing either the UA entirely or add IG as referer, the download speed instantly becomes 10MB/s (my bandwidth limit).

One can easily test it by adding 

```py
            anonymous_session.headers['user-agent'] = None
```

after L509 at `instaloadercontext.py`:

https://github.com/instaloader/instaloader/blob/8c1129e5563490b3cfe37f8d172c80907452da07/instaloader/instaloadercontext.py#L509-L511

**Comment by fireattack** [comment]
Just learned from others that if you add IG as referer, no speed limit either. Updated the OP accordingly.

**Comment by aandergr** [comment]
Nice finding! This looks like a promising fix for #1145. Would you create a pull request with the proposed changes?
